mithril-common/src/entities/signer.rs

```diff
@@ -28,20 +63,60 @@ impl Signer {
         let mut hasher = Sha256::new();
         hasher.update(self.party_id.as_bytes());
         hasher.update(self.verification_key.as_bytes());
+        if let Some(verification_key_signature) = &self.verification_key_signature {
+            hasher.update(verification_key_signature.as_bytes());
+        }
+        if let Some(operational_certificate) = &self.operational_certificate {
+            hasher.update(operational_certificate.as_bytes());
+        }
         hex::encode(hasher.finalize())
     }
 }
 
+impl From<SignerWithStake> for Signer {
+    fn from(other: SignerWithStake) -> Self {
+        Signer::new(
+            other.party_id,
+            other.verification_key,
+            other.verification_key_signature,
+            other.operational_certificate,
+            other.kes_period,
+        )
+    }
+}
+
 /// Signer represents a signing party in the network (including its stakes)
 #[derive(Clone, Debug, PartialEq, Eq, Default, Serialize, Deserialize)]
 pub struct SignerWithStake {
     /// The unique identifier of the signer
+    // TODO: Should be removed once the signer certification is fully deployed
     #[serde(rename = "party_id")]
     pub party_id: PartyId,
 
     /// The public key used to authenticate signer signature
     #[serde(rename = "verification_key")]
-    pub verification_key: String,
+    pub verification_key: HexEncodedKey,
+
+    /// The encoded signer 'Mithril verification key' signature (signed by the Cardano node KES secret key)
+    // TODO: Option should be removed once the signer certification is fully deployed
+    #[serde(
+        rename = "verification_key_signature",
```

Comment:
Same as comment above, why rename if the field is named the same?

Reply:
Done :heavy_check_mark: